packages/core/src/model/nodes/ChatNodeBase.ts

```diff
@@ -1452,24 +1452,30 @@ export const ChatNodeBase = {
           maxTimeout: 5000,
           randomize: true,
           signal: context.signal,
-          onFailedAttempt(err) {
-            if (err.toString().includes('fetch failed') && err.cause) {
+          onFailedAttempt(originalError) {
+            let err = originalError;
+            if (originalError.toString().includes('fetch failed') && originalError.cause) {
               const cause =
-                getError(err.cause) instanceof AggregateError
-                  ? (err.cause as AggregateError).errors[0]
-                  : getError(err.cause);
+                getError(originalError.cause) instanceof AggregateError
+                  ? (originalError.cause as AggregateError).errors[0]
+                  : getError(originalError.cause);
 
               err = cause;
             }
 
-            context.trace(`ChatNode failed, retrying: ${err.toString()}`);
-
             if (context.signal.aborted) {
               throw new Error('Aborted');
             }
 
+            context.trace(`ChatNode failed, retrying: ${err.toString()}`);
+
             const { retriesLeft } = err;
 
+            // Retry network errors
+            if (err.toString().includes('terminated') || originalError.toString().includes('terminated') || err.toString().includes('fetch failed')) {
```

Comment:
This doesn't seem like a great way to catch network errors, but fetch doesn't provide a unique error class for this. Also not sure if we should match `fetch failed` against `originalError` - it seems originalError in this case would never be an `OpenAIError` but we should probably want to retry it?

Reply:
seems... okay I spose?